lib/buckets/common.js

```diff
@@ -53,8 +58,21 @@ var resemblesIpRegexStr = format('^%s\.%s\.%s\.%s$', threeDigitRegexStr,
 var resemblesIpRegex = new RegExp(resemblesIpRegexStr);
 
 function isValidBucketName(name) {
-    return bucketRegex.test(name) && !resemblesIpRegex.test(name) &&
-        name.length >= 3 && name.length <= 63;
+    assert.string(name, 'name');
+
+    return (bucketRegex.test(name) && !resemblesIpRegex.test(name) &&
+        !nullRegex.test(name) &&
+        name.length >= 3 && name.length <= 63);
+}
+
+/*
+ * A valid bucket object name contains characters with valid UTF-8 characters
+ * (non-null) and is a maximum of 1024 characters.
+ */
+function isValidBucketObjectName(name) {
+    assert.string(name, 'name');
+
+    return (!nullRegex.test(name) && name.length > 0 && name.length <= 1024);
```

Comment:
- Can we pull this 1024 max len out to a constant?

- Do we want to consider copying S3 which specifies the max length is 1024 *bytes*? Limiting on bytes means you have a physical limit (if theoretically we were calculating DB space requirements). A unicode char, practically speaking, can be 4 bytes in UTF-8, so a 1024 *char* limit would mean a 4096 byte potential upper limit.


```javascript
$ node
> s1 = 'hi'
'hi'
> s2 = '\u{1F436}\u{1F436}'
'🐶🐶'
> s1.length
2
> s2.length
4
> Buffer.byteLength(s1)
2
> Buffer.byteLength(s2)
8
```

If we copy S3, we should include a ref to their docs: https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html#object-keys

Reply:
- Agreed, I can pull out 1024 and 63 above and make them constants so they aren't as hidden
- Our documentation (linked above) currently states "characters" and not "bytes", so at the very least our documentation is inline with our current implementation.  However, I personally lean more towards preferring `Buffer.byteLength(s)` as opposed to `s.length` as using byte size is less surprising when it comes to actually storing the data.

@kellymclaughlin do you have thoughts on this? I'm leaning towards changing our restrictions to be bytes and not character length.

Reply:
Yeah, I agree. I think that's a good idea.